### DISCUSSION TITLE
Import tables from Excel, Google Sheets, SQL and ect.

### DISCUSSION THREAD

**Discussion by RC441** [jgraph/drawio#4270] [General]
Hi, I need to import pairwise tables. Since it's big tables (in most cases) i need to be able to import tables either by copy paste or through the menu. Also it would be imported thought SQL commands. For example, here https://sqlizer.io/#/ it is possible to convert excel table to SQL commands. Thanks!

**Comment by Peeeet** [comment]
There is this https://drawio-app.com/blog/sql-plugin/

and also you can import SQL and CSV - plus a config section to define how the CSV rows and columns would be turned into DrawIO nodes.  This is available under the "Arrange" menu > Insert > Advanced > SQL... or CSV...

The SQL example that pops up creates some tables, with a little tweaking you can probably get it to make the kind of tables you are looking for.

**Comment by lastlink** [comment]
@125401qwerty I have a pr #4122 with improvements to the sql plugin that you have to install yourself manually (one file change) for sql import into uml draw io diagram
